public/app/plugins/datasource/prometheus/language_provider.ts

```
@@ -111,11 +115,54 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     return defaultValue;
   };
 
+  requestViaWorker(url: string, hasObjectResponse?: false): Promise<string[]>;
+  requestViaWorker(url: string, hasObjectResponse?: true): Promise<PromMetricsMetadata>;
+  requestViaWorker(url: string, hasObjectResponse?: boolean): Promise<string[] | PromMetricsMetadata>;
+  requestViaWorker(url: string, hasObjectResponse = false): Promise<string[] | PromMetricsMetadata> {
+    return new Promise(resolve => {
+      let nodes: string[] | PromMetricsMetadata = hasObjectResponse ? {} : [];
+      const streamWorker = new (StreamJSONResponse as any)() as StreamJSONResponseWorker;
+
+      streamWorker.onmessage = e => {
+        if (e.data === JSON_STREAM_DONE) {
+          streamWorker.terminate();
+          resolve(nodes);
+        }
+
+        if (Array.isArray(nodes)) {
+          nodes = nodes.concat(e.data);
+        } else {
+          Object.assign(nodes, e.data);
+        }
+      };
+
+      streamWorker.onerror = error => {
+        resolve(nodes);
```

Shouldn't we reject the promise with error in that case? Maybe having an Error with nodes property, so that component can then decide what to do with it, like ideal I would assume show an error toast while updating the metrics so app can continue working but at least user knows not everything is fine.

---

I was just working off of the previous implementation - which was to just return the default if an error happens and the error (such as 405 with prometheus metadata) only gets logged to the console.

---

Got it but still, we should do something with the error, right now we are just silently ignoring it, which is the most frustrating thing when debugging some weird behavior.

---

Actually, the button does say "(No metadata found)".